system_test/helpers/aest_docker.erl

```diff
@@ -197,18 +198,23 @@ start_node(#{container_id := ID, hostname := Name} = NodeState) ->
     log(NodeState, "Container ~p [~s] started", [Name, ID]),
     NodeState.
 
--spec stop_node(node_state()) -> node_state().
-stop_node(NodeState) -> stop_node(NodeState, #{}).
-
 -spec stop_node(node_state(), stop_node_options()) -> node_state().
 stop_node(#{container_id := ID, hostname := Name} = NodeState, Opts) ->
     Timeout = maps:get(soft_timeout, Opts, ?EPOCH_STOP_TIMEOUT),
     case is_running(ID) of
         false ->
             log(NodeState, "Container ~p [~s] already not running", [Name, ID]);
         true ->
-            aest_docker_api:exec(ID, ["/home/epoch/node/bin/epoch", "stop"]),
-            case wait_stopped(ID, Timeout) of
+            Cmd = ["/home/epoch/node/bin/epoch", "stop"],
+            CmdStr = lists:join($ , Cmd),
+            log(NodeState,
+                "Container ~p [~s] still running: "
+                "attempting to stop node by executing command ~s",
+                [Name, ID, CmdStr]),
+            aest_docker_api:exec(ID, Cmd, #{timeout => Timeout}),
+            log(NodeState, "Command executed on container ~p [~s]: ~s",
+                [Name, ID, CmdStr]),
+            case wait_stopped(ID, Timeout) of %% TODO Fix this call that has timeout actual parameter as seconds but handled inside function definition as milliseconds.
```

Comment:
It looks like this does not stop container so then call to delete node fails with "You cannot remove a running container" hence leaving container behind. This point would benefit from refactoring.

Reply:
Actually the problem is that the exec is executed with too small timeout so this is not executed.

Reply:
This was less visible before as the timeout was taken from input but hardcoded in HTTP call.

Reply:
Addressed in new commit.